### ISSUE TITLE
Suspicious non-const static buffer in krb5 get_in_tkt.c set_ptypes

### ISSUE THREAD

**Issue by riastradh** [heimdal/heimdal#1133]
**Describe the bug**
https://github.com/heimdal/heimdal/blob/9cbafd7320b8023ea8c6ea89643803bd021adda5/lib/krb5/get_in_tkt.c#L316-L319
https://github.com/heimdal/heimdal/blob/9cbafd7320b8023ea8c6ea89643803bd021adda5/lib/krb5/get_in_tkt.c#L321
https://github.com/heimdal/heimdal/blob/9cbafd7320b8023ea8c6ea89643803bd021adda5/lib/krb5/get_in_tkt.c#L337

Unclear if there's anything actually wrong with this but it's suspicious to return a non-const pointer to a static buffer to a caller.

**To Reproduce**
code inspection

**Comment by nicowilliams** [comment]
Indeed, this is a bug, though in a deprecated function's code path.
